chromadb/api/types.py

```diff
@@ -122,11 +115,177 @@ def is_image(target: Any) -> bool:
     return True
 
 
-def maybe_cast_one_to_many_image(target: OneOrMany[Image]) -> Images:
-    if is_image(target):
-        return cast(Images, [target])
-    # Already a sequence
-    return cast(Images, target)
+class BaseRecordSet(TypedDict):
+    """
+    The base record set includes 'data' fields and embeddings.
+    """
+
+    embeddings: Optional[Embeddings]
+    documents: Optional[Documents]
+    images: Optional[Images]
+    uris: Optional[URIs]
+
+
+def get_default_embeddable_record_set_fields() -> Set[str]:
+    """
+    Returns the set of fields that can be embedded on a Record Set.
+    This is a way to avoid hardcoding the fields in multiple places,
+    and keeps them immutable.
+    """
+    return {"documents", "images", "uris"}
+
+
+class InsertRecordSet(BaseRecordSet):
+    """
+    A set of records for inserting.
+    """
+
+    ids: IDs
+    metadatas: Optional[Metadatas]
+
+
+def unpack_base_record_set(
```

Comment:
qq  - why is this called unpack? I view it as packing lol

Reply:
copilot tried to fight me about this too, I think of it as 'unpack' because it's where the `OneOrMany` stuff happens.

Reply:
maybe "normalize" is better